cmd/decode.go

```diff
@@ -95,21 +104,71 @@ func validateAndRun() error {
 	return run(options.metaOnly, outMediaType, in, os.Stdout)
 }
 
+// Original requirement from etcd WAL log analysis tool etcd-dump-logs,
+// to serve as a decoder to decode k8s objects from each entry of the etcd WAL log.
+// Read data from os.Stdin, decode object line by line, and print out in os.Stdout
+func runInBatchMode(metaOnly bool, outMediaType string, out io.Writer) error {
+	var err error
+	batchmode := true
+	inputReader := bufio.NewReader(os.Stdin)
+	for {
+		input, err := inputReader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			fmt.Printf("EOF\n")
```

Comment:
Why print EOF?

Reply:
Also, we should write to `out` using `fmt.Fprintf(out, ...` or `out.Write`, not `fmt.Printf` in this function. This allows auger to be consistent in how output is handled.